libs/hook-dapp-lib/src/utils.ts

```diff
@@ -19,28 +18,24 @@ export function matchHooksToDapps(hooks: CowHook[], dapps: HookDappBase[]): Hook
     {} as Record<string, HookDappBase | undefined>,
   )
 
-  return hooks.map((hook) => {
-    const dapp = dappsMap[hook.callData.slice(-HOOK_DAPP_ID_LENGTH)]
+  return (
+    hooks
+      // Skip hooks before the hooks store was introduced
+      .filter((hook) => !!hook.dappId)
```

Comment:
Thinking back to how you had before, I liked how it could detect permit hooks used in the past.

My comment was more about the casting to latest appData than the logic itself.

Showing permit hooks for past orders is a nice extra feature IMO.

Reply:
Yeah, I was thinking about that.
But I'm not sure does it worth or not. From the code perspective it would be an eyesore.

Reply:
@alfetopito I hope you don't mind if I merge the PR.
Anyway, if you feel the feature (detect "old" permit hook) worth it - just tell me, I will add it back